e2e/benchmarks/local-benchmark/dump.js

```diff
@@ -0,0 +1,235 @@
+  }
+  // In
+  // https://storage.googleapis.com/tfhub-tfjs-modules/mediapipe/tfjs-model/face_landmarks_detection/attention_mesh/1/model.json,
+  // some inputs are prefixed with '^'.
+  if (!inputs || inputs.length == 0 || inputs[0].startsWith('^')) {
+    return null;
+  }
+
+  let tensorMap = {};
+  for (let i = 0; i < inputs.length; i++) {
+    const key = inputs[i].split(':')[0];
+    if (dumpedJson[key] == null || dumpedJson[key][0] == null) {
+      console.warn('Tensor ' + key + ' is null!');
+      return null;
+    }
+    const tensorInfo = dumpedJson[key][0];
+    const tensor = tf.tensor(
+        Object.values(tensorInfo.value[0]), tensorInfo.shape, tensorInfo.dtype);
+    tensorMap[key] = tensor;
+  }
+
+  return tensorMap;
+}
+
+async function predictOp(
+    model, modelJson, dumpedJson, outputNodeName, backend) {
+  await tf.setBackend(backend);
+  const tensorMap =
+      await createTensorMap(outputNodeName, modelJson, dumpedJson);
+  if (tensorMap == null) {
+    return null;
+  }
+  let prediction;
+  try {
+    // TODO(#6861): Support tensor with type conversion.
+    prediction = await model.executeAsync(tensorMap, outputNodeName);
+  } catch (e) {
+    console.warn(e.message);
+    return null;
+  }
+
+  let intermediateData =
+      await getIntermediateTensorInfo(model.getIntermediateTensors());
+  await getPredictionData(prediction);
+  model.disposeIntermediateTensors();
+  return intermediateData;
+}
+
+async function dumpOp(model, dumpedJson, backends, outputNodeName, index) {
+  let objects1 = {};
+  let objects2 = {};
+  const modelJson = model.artifacts;
+  const referenceObject = await predictOp(
+      model, modelJson, dumpedJson, outputNodeName, backends[1]);
+
+  const predictObject = await predictOp(
+      model, modelJson, dumpedJson, outputNodeName, backends[0]);
+  if (predictObject && referenceObject) {
+    objects1 = {index, ...predictObject[`${outputNodeName}`]};
+    objects2 = {index, ...referenceObject[`${outputNodeName}`]};
+  }
+  return [objects1, objects2];
+}
+
+
+/**
+ * Dump the predict results of two backends and save diffs to files.
+ * @param model The loaded model.
+ * @param jsonObjects The predict results from backends.
+ * @param backends [predict backend, reference backend].
+ * @param benchmark Used for generating dump file name.
+ * @param level 0, dump close diffs. 1, dump any diffs. 2, dump all tensors.
+ * @param length Used for controlling how many tensors will be dumped. -1 dump
+ *     all.
+ */
+async function dump(
+    model, jsonObjects, backends, benchmark = '', level = 0, length = 1) {
+  const jsonObject1 = jsonObjects[0];
+  const jsonObject2 = jsonObjects[1];
+  const dumpObjects1 = {};
+  const dumpObjects2 = {};
+  const keys = Object.keys(jsonObjects[0]);
+  const prefix = `dump_${benchmark}_${level}`;
+  let dumpCount = 0;
+  for (let i = 0; i < keys.length; i++) {
+    const key = keys[i];
+    if (compareData(jsonObject1[key], jsonObject2[key], level)) {
```

Comment:
If the reference and predict tensors are different, and level is "dump all", will you do the check again. I suggest we always recheck the different ones.

Reply:
As synced offline, dump all is removed